### PR TITLE
Better handling of block doc comments

### LINKED ISSUE #7774: Block doc comments are not recognized properly
This is related to #6873, but I've decided to post a new issue instead of a comment, since that only addresses syntax highlighting. I played around with it a bit, and it seems like block doc comments straight up aren't recognized properly.

Capture

Also, indented block comments don't render correctly in tooltips.

Capture

### PR DESCRIPTION
**Pull request by Veykril**
Moves doc string processing to `Attrs::docs`, as we need the indent info from all comments before being able to know how much to strip

Closes #7774

### CODE REVIEW

File: crates/hir_def/src/attr.rs [hunk 1]
```diff
@@ -77,33 +77,19 @@ impl RawAttrs {
     pub(crate) const EMPTY: Self = Self { entries: None };
 
     pub(crate) fn new(owner: &dyn AttrsOwner, hygiene: &Hygiene) -> Self {
```
Comment (matklad) [inline crates/hir_def/src/attr.rs:79]:
```suggestion
    pub(crate) fn new(owner: &dyn ast::AttrsOwner, hygiene: &Hygiene) -> Self {
```

**Comment by Veykril** [comment]
bors retry

**Comment by Veykril** [comment]
changelog internal handle block doc comments better

**Pull request merged**